File: matlab/vts_wrapper/vts/structures/MonteCarlo/DetectorInputs/TallyDetails.m
classdef TallyDetails
    methods (Static)
        function details = Default()
            details.IsReflectanceTally = false;
            details.IsTransmittanceTally = false;
            details.IsSpecularReflectanceTally = false;
            details.IsInternalSurfaceTally = false;
            details.IspMCReflectanceTally = false;
            details.IsDosimetryTally = false;
            details.IsVolumeTally = false;
            details.IsCylindricalTally = false;
            details.IsNotImplementedForDAW = false;
            details.IsNotImplementedForCAW = false;
            details.IsNotImplementedYet = false;
        end
        function details = FromDetailsNET(detailsNET)            
            details.IsReflectanceTally = detailsNET.IsReflectanceTally;
            details.IsTransmittanceTally = detailsNET.IsTransmittanceTally;
            details.IsSpecularReflectanceTally = detailsNET.IsSpecularReflectanceTally;
            details.IsInternalSurfaceTally = detailsNET.IsInternalSurfaceTally;
            details.IspMCReflectanceTally = detailsNET.IspMCReflectanceTally;
            details.IsDosimetryTally = detailsNET.IsDosimetryTally;
            details.IsVolumeTally = detailsNET.IsVolumeTally;
            details.IsCylindricalTally = detailsNET.IsCylindricalTally;
            details.IsNotImplementedForDAW = detailsNET.IsNotImplementedForDAW;
            details.IsNotImplementedForCAW = detailsNET.IsNotImplementedForCAW;
            details.IsNotImplementedYet = detailsNET.IsNotImplementedYet;
        end        
        function detailsNET = ToDetailsNET(details)
            detailsNET = Vts.MonteCarlo.Detectors.TallyDetails;
            detailsNET.IsReflectanceTally = details.IsReflectanceTally;
            detailsNET.IsTransmittanceTally = details.IsTransmittanceTally;
            detailsNET.IsSpecularReflectanceTally = details.IsSpecularReflectanceTally;
            detailsNET.IsInternalSurfaceTally = details.IsInternalSurfaceTally;
            detailsNET.IspMCReflectanceTally = details.IspMCReflectanceTally;
            detailsNET.IsDosimetryTally = details.IsDosimetryTally;
            detailsNET.IsVolumeTally = details.IsVolumeTally;
            detailsNET.IsCylindricalTally = details.IsCylindricalTally;
            detailsNET.IsNotImplementedForDAW = details.IsNotImplementedForDAW;
            detailsNET.IsNotImplementedForCAW = details.IsNotImplementedForCAW;
            detailsNET.IsNotImplementedYet = details.IsNotImplementedYet;
        end
    end
end

File: src/Vts/MonteCarlo/Detectors/TallyDetails.cs
namespace Vts.MonteCarlo.Detectors
{
    /// <summary>
    /// details for each detector that identifies type of detector
    /// </summary>
    public class TallyDetails
    {
        /// <summary>
        /// Boolean identifying all reflectance tallies
        /// </summary>
        public bool IsReflectanceTally { get; set; }
        /// <summary>
        /// Boolean identifying all transmittance tallies
        /// </summary>
        public bool IsTransmittanceTally { get; set; }
        /// <summary>
        /// Boolean identifying all specular reflectance tallies
        /// </summary>
        public bool IsSpecularReflectanceTally { get; set; }
        /// <summary>
        /// Boolean identifying all internal surface tallies
        /// </summary>
        public bool IsInternalSurfaceTally { get; set; }
        /// <summary>
        /// Boolean identifying all perturbation MC reflectance tallies
        /// </summary>
        public bool IspMCReflectanceTally { get; set; }
        /// <summary>
        /// Boolean identifying all perturbation MC transmittance tallies
        /// </summary>
        public bool IspMCTransmittanceTally { get; set; }
        /// <summary>
        /// Boolean identifying all bounding volume tallies
        /// </summary>
        public bool IsLateralBoundingVolumeTally { get; set; }
        /// <summary>
        /// Boolean identifying all volume tallies
        /// </summary>
        public bool IsVolumeTally { get; set; }
        /// <summary>
        /// Boolean identifying whether detector is in cylindrical coordinates
        /// </summary>
        public bool IsCylindricalTally { get; set; }
        /// <summary>
        /// Boolean identifying whether detector is not implemented yet for DAW
        /// </summary>
        public bool IsNotImplementedForDAW { get; set; }
        /// <summary>
        /// Boolean identifying whether detector is not implemented yet for CAW
        /// </summary>
        public bool IsNotImplementedForCAW { get; set; }
        /// <summary>
        /// Boolean identifying whether detector is not implemented yet
        /// </summary>
        public bool IsNotImplementedYet { get; set; }
    }
}
